ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -248,8 +249,17 @@ run runargs@RunNodeArgs{..} =
           nodeKernel
           rnTraceNTN
           (NTN.defaultCodecs codecConfig version)
-          (Just 70) -- timeout after waiting this long for the next header
-                    -- 70s allows for 3 slots (3 * 20s)
+          -- These values approximately correspond to false positive thresholds
+          -- for streaks of empty slots with 99% probability, 99.9% probability up to
+          -- 99.999% probability.
+          -- t = T_s [log (1-Y) / log (1-f)]
+          -- Y = [0.99, 0.999...]
+          -- T_s = slot length of 1s.
+          -- f = 0.05
+          -- The timeout is randomly picked per bearer to avoid all bearers going down
+          -- at the same time in case of a long streak of empty slots.
+          -- TODO: workaround until peer selection governor.
+          (generateElement [90, 135, 180, 224, 269])
```

Comment:
Don't use that implementation of `generateElement`. It uses system entropy. Use an impl based on a simple PRNG e.g. `System.Random.randomRIO`
```
randomElem xs = do
  ix <- randomRIO (0, length xs - 1)
  return (xs !! ix)
```

Reply:
Updated with a randomRIO.